af_ktls.c

```diff
@@ -984,10 +988,14 @@ static inline ssize_t tls_peek_data(struct tls_sock *tsk, unsigned flags)
 			goto peek_failure;
 		}
 
+		if (peeked_size < KTLS_MIN_RECORD_SIZE(tsk)) {
+			ret = -EBADF;
```

Comment:
It doesn't look correct to return EBADF here - we just might not have enough bytes for a full message (if for example a tcp message was split in the first 5 header bytes).  I think we should just return without error?